keylime/tpm/tpm_main.py

```diff
@@ -1051,6 +1051,64 @@ def __tpm2_checkquote(self, pubaik, nonce, quoteFile, sigFile, pcrFile, hash_alg
         retDict = self.__run(command, lock=False)
         return retDict
 
+    def __tpm2_printquote(self, quoteFile):
+        command = ["tpm2_print", "-t", "TPMS_ATTEST", quoteFile]
+        retDict = self.__run(command, lock=False)
+        return retDict
+
+    def _tpm2_printquote(self, quote, compressed):
+        """Get TPM timestamp info from quote
+        :param quote: quote data in the format 'r<b64-compressed-quoteblob>:<b64-compressed-sigblob>:<b64-compressed-pcrblob>
+        :param compressed: if the quote data is compressed with zlib or not
+        :returns: Returns the 'retout' from running tpm2_print and True in case of success, None and False in case of error.
+        This function throws an Exception on bad input.
+        """
+
+        if quote[0] != "r":
+            raise Exception(f"Invalid quote type {quote[0]}")
+        quote = quote[1:]
+
+        quote_tokens = quote.split(":")
+        if len(quote_tokens) < 3:
+            raise Exception(f"Quote is not compound! {quote}")
+
+        quoteblob = base64.b64decode(quote_tokens[0])
+
+        if compressed:
+            logger.warning("Decompressing quote data which is unsafe!")
+            quoteblob = zlib.decompress(quoteblob)
+
+        qfd = -1
+        quoteFile = None
+
+        try:
+            # write out quote
+            qfd, qtemp = tempfile.mkstemp()
+            with open(qtemp, "wb") as quoteFile:
+                quoteFile.write(quoteblob)
+
+            retDict = self.__tpm2_printquote(quoteFile.name)
+            retout = retDict["retout"]
+            reterr = retDict["reterr"]
+            code = retDict["code"]
+        except Exception as e:
+            logger.error("Error printing quote: %s", str(e))
+            logger.exception(e)
+            return None, False
```

Comment:
We should avoid Go error idioms here. Return `retout` if all is OK, and return None if there is something wrong.  If is too wrong maybe even re-raise the exception and manage it in upper levels.

Reply:
Again, same as above. I agree with you both, but for this PR, I have tried to follow the pattern already on the code for other commands. Not opposed to change it for **all** commands in a subsequent PR, dedicated to "code improvement/technical debt reduction"

Reply:
I'm also ok if we open a ticket to track this tech debt reduction and letting this PR move forward.